modules/users/server/models/user.server.model.js

```diff
@@ -112,7 +112,7 @@ UserSchema.pre('save', function(next) {
  */
 UserSchema.methods.hashPassword = function(password) {
 	if (this.salt && password) {
-		return crypto.pbkdf2Sync(password, this.salt, 10000, 64).toString('base64');
+		return crypto.pbkdf2Sync(password, new Buffer(this.salt, 'base64'), 10000, 64).toString('base64');
```

Comment:
@ilanbiala It is necessary to be explicit about encoding and decoding the buffer using base64 in both cases.

I'm not worried about the overhead of re-instantiating the buffer here, because crypto.pbkdf2Sync() will create a buffer anyway if you pass a string as the salt, but it will make the incorrect assumption that the string contains binary-encoded data.

The buffer will always need to be re-instantiated somehow, because the setter created by the Mongoose model would automatically convert the buffer to a string when it's assigned to this.salt on L103.